pdal/compression/LazPerfVlrCompression.cpp

```diff
@@ -206,23 +205,61 @@ void LazPerfVlrCompressor::done()
 class LazPerfVlrDecompressorImpl
 {
 public:
-    LazPerfVlrDecompressorImpl(std::istream& stream, const char *vlrData,
-        std::streamoff pointOffset) :
-        m_stream(stream), m_inputStream(stream), m_chunksize(0),
-        m_chunkPointsRead(0)
+    LazPerfVlrDecompressorImpl(std::istream& stream, int format, int ebCount,
+            std::streamoff pointOffset, const char *vlrdata) :
+        m_stream(stream), m_fileStream(stream), m_format(format), m_ebCount(ebCount),
+        m_chunkPointsRead(0), m_vlr(vlrdata)
     {
-        laszip::io::laz_vlr zipvlr(vlrData);
-        m_chunksize = zipvlr.chunk_size;
-        m_schema = laszip::io::laz_vlr::to_schema(zipvlr);
-        m_stream.seekg(pointOffset + sizeof(int64_t));
+        m_stream.seekg(pointOffset);
+        ILeStream in(&stream);
+
+        uint64_t chunkTablePos;
+        in >> chunkTablePos;
+
+        in.seek(chunkTablePos);
+        uint32_t version;
+        uint32_t numChunks;
+        in >> version;
+        in >> numChunks;
+
+        assert(version == 0);
```

Comment:
Should this be a `pdal_error` instead?  I'm not quite clear on what the `version` means in this context so it could certainly be fine as-is.  But if version `1` could come out sometime in the future with a different expected behavior, then a runtime error would seem appropriate.

Reply:
I might have put that in for my own sanity? It's Martin's versioning thing for the LASzip VLR.

I think your idea is a good one if I can find an easy way to do it.